app/models/exam.rb

```diff
@@ -131,8 +131,8 @@ def attempts_left_for(assignment)
     max_attempts_for(assignment) - (assignment.attempts_count || 0)
   end
 
-  def limited?
-    true
+  def limited_for?(exercise)
+    max_attempts_for(exercise).present?
```

Comment:
do we have a name for the exercise/assignment interface? because this method says it takes an assignment, but could actually take either

Reply:
Hmm, the problem is that it should take only one of them. Which objects are we actually passing?